src/odoc/interface.mld

```diff
@@ -0,0 +1,95 @@
+- A few extra tags are supported:
+  + [@returns] is a synonym for [@return]
+  + [@raises] is a synonym for [@raise]
+  + [@open] and [@closed] and [@inline] are hints for how 'included' signatures should be rendered
+  + [@canonical] allows a definition of a module to be marked as canonically elsewhere. 
+
+{3 Reference syntax}
+Odoc has a far more powerful reference resolution mechanism than ocamldoc. While it supports the mechanism in ocamldoc used for disambiguating between different types of references,
+it offers a more powerful alternative. The new mechanism allows for disambiguation of each part in a dotted reference rather than just the final part. For example, 
+where in the reference manual it suggests the syntax [{!type:Foo.Bar.t}] to designate a type, and [{!val:Foo.Bar.t}] a value of the same name, the new Odoc syntax for these
+comments would be [{!Foo.Bar.type-t}] and [{!Foo.Bar.val-t}]. This allows odoc to disambiguate when there are other ambiguous elements within the path. For example, we can
+distinguish between a type or value t within a module or module type with the same name: [{!module-Foo.module-type-Bar.type-t}] or [{!module-type-Foo.module-Bar.val-t}].
+
+Additionally we support extra annotations:
+- [module-type] is a replacement for [modtype]
+- [class-type] is a replacement for [classtype]
+- [exn] is recognised as [exception]
+- [extension] refers to a type extension
+- [field] is a replacement for [recfield]
+- [instance-variable] refers to instance variables
+- [label] refers to labels introduced in anchors
+- [page] refers to [mld] pages as outlined above
+- [value] is recognised as [val]
+
+This will be described more completely in a separate document.
+
+{2 CLI interface}
+
+The way in which the odoc CLI is invoked is not trivial, and requires careful
+ordering and correct arguments to produce correctly linked documentation. It is not expected that
+end-users will invoke odoc by hand, but rather it will be driven by a separate tool. As a consequence of
+this it is important that we preserve the ability of these tools to create good documentation with
+each release of odoc, and thus we will ensure backward compatibility of the CLI as much as possible.
+There are currently 3 tools that 'drive' odoc that are considered 'first class' in that we will not make
+releases of odoc whilst knowingly breaking these tools. These are:
+
+- odig
+- dune
+- ocaml
+
+Ocaml here refers to the newly merged configure option to build the standard library documentation with
+odoc. If the recommended way of invoking odoc changes we will work with the maintainers of these projects
+to ensure they are updated correspondingly.
+
+Additionally there will be a reference implementation of a tool to build Odoc's documentation which should
+serve as a guide for anyone building other 'drivers' of odoc.
+
+{2 Output formats}
+
+Odoc currently outputs HTML files, man pages and latex documents. In a similar vein to the CLI interface,
```

Comment:
from the odoc dev meeting: need a statement of intermediate file formats as well (e.g. are odoc files supported)

Reply:
done